packages/wallets/src/wallets/wallet-factory.ts

```diff
@@ -257,6 +268,48 @@ export class WalletFactory {
             }
             compareSignerConfigs(adminSignerArgs, existingWalletSigner);
         }
+
+        if (args.delegatedSigners != null && args.chain === "solana") {
+            this.validateDelegatedSigners(existingWallet, args.delegatedSigners);
+        }
+    }
+
+    private validateDelegatedSigners(
+        existingWallet: GetWalletSuccessResponse,
+        inputDelegatedSigners: Array<DelegatedSigner>
+    ): void {
+        const existingDelegatedSigners = (existingWallet?.config as any)?.delegatedSigners as
+            | DelegatedSignerResponse[]
+            | undefined;
+
+        if (existingDelegatedSigners == null || existingDelegatedSigners.length === 0) {
+            if (inputDelegatedSigners.length > 0) {
+                throw new WalletCreationError(
+                    `${inputDelegatedSigners.length} delegated signer(s) specified, but wallet "${existingWallet.address}" has no delegated signers. ${DELEGATED_SIGNER_MISMATCH_ERROR}`
+                );
+            } else {
+                return; // Both wallet and args have no delegated signers, no further validation needed
+            }
+        }
+
+        if (inputDelegatedSigners.length !== existingDelegatedSigners.length) {
+            throw new WalletCreationError(
+                `${inputDelegatedSigners.length} delegated signer(s) specified, but wallet "${existingWallet.address}" has ${existingDelegatedSigners.length} delegated signers. ${DELEGATED_SIGNER_MISMATCH_ERROR}`
+            );
+        }
+
+        for (const argSigner of inputDelegatedSigners) {
+            const matchingExistingSigner = existingDelegatedSigners.find(
+                (existingSigner) => existingSigner.locator === argSigner.signer
+            );
+
+            if (matchingExistingSigner == null) {
+                const walletSigners = existingDelegatedSigners.map((s) => s.locator).join(", ");
+                throw new WalletCreationError(
```

Comment:
This error does make sense. If the input does not have delegatedSigners and a wallet exists with them, should be okay. If the input has them, they need to match. Not match exactly though, because new signers could have been added after the fact